sqlite.go

```diff
@@ -490,13 +490,23 @@ func (s *stmt) ExecContext(ctx context.Context, args []driver.NamedValue) (drive
 	if err := s.bindAll(args); err != nil {
 		return nil, s.reserr("Stmt.Exec(Bind)", err)
 	}
+	done := make(chan struct{})
 	if ctx.Value(queryCancelKey{}) != nil {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithCancel(ctx)
-		defer cancel()
+		pctx, pcancel := context.WithCancel(ctx)
+		defer pcancel()
 
 		db := s.stmt.DBHandle()
-		go func() { <-ctx.Done(); db.Interrupt() }()
+		stop := context.AfterFunc(pctx, func() {
+			defer close(done)
+			if ctx.Err() != nil {
+				db.Interrupt()
+			}
+		})
+		// In the event we get an error from the initial step and exit early,
```

Comment:
s/step/StepResult/ to clarify that "step" here means the sqlite3_step and not "initial step" in the general sense? (took me a second to parse it correctly)

But don't we want to stop this regardless, even in the ... oh! The normal flow is "defer pcancel()" to make the AfterFunc fire, which then does the close(done), when 521 blocks on. Might be worth a little comment like:

     defer pcancel() // to make the AfterFunc fire and close(done)

Reply:
Good idea, thanks.